client/js/components/user/DpUserFormFields.vue

```diff
@@ -252,9 +251,13 @@ export default {
      * - user is set: roles for current organisation
      */
     allowedRolesForOrga () {
-      return (this.currentUserOrga.id === '')
-        ? this.rolesInRelationshipFormat
-        : Object.values(this.$store.state.orga.items[this.currentUserOrga.id].relationships.allowedRoles.list())
+      let allowedRoles = this.rolesInRelationshipFormat
+      if (this.currentUserOrga.id !== '' && hasOwnProp(this.organisations[this.currentUserOrga.id].relationships, 'allowedRoles')) {
+        allowedRoles = Object.values(this.organisations[this.currentUserOrga.id].relationships.allowedRoles.list())
+      } else if (this.currentUserOrga.id !== '' && !hasOwnProp(this.organisations[this.currentUserOrga.id].relationships, 'allowedRoles')) {
+        allowedRoles = this.getOrgaAllowedRoles(this.currentUserOrga.id)
+      }
+      return allowedRoles
```

Comment:
nitpick:
I hope this is the same logic but rearranged (to be shorter). Even not sure if its really better
```suggestion
      let allowedRoles

      if (this.currentUserOrga.id === '' ) {
          allowedRoles = this.rolesInRelationshipFormat
      } else if (hasOwnProp(this.organisations[this.currentUserOrga.id].relationships, 'allowedRoles')) {
        allowedRoles = Object.values(this.organisations[this.currentUserOrga.id].relationships.allowedRoles.list())
      } else {
        allowedRoles = this.getOrgaAllowedRoles(this.currentUserOrga.id)
      }
      
      return allowedRoles
```

Reply:
thx! This logic is better and works fine. Done here https://github.com/demos-europe/demosplan-core/pull/750/commits/51d180a277b07fb9a3561b0093e0500bb086aee1